gtda/diagrams/features.py

```diff
@@ -642,27 +642,28 @@ def fit(self, X, y=None):
         _homology_dimensions_counts = dict(zip(homology_dimensions_fit,
                                                counts))
         if self.n_coefficients is None:
-            self.n_coefficients_ = [_homology_dimensions_counts[dim]
-                                    for dim in self.homology_dimensions_]
-        elif type(self.n_coefficients) == list:
+            self.n_coefficients_ = \
+                tuple([_homology_dimensions_counts[dim]
+                       for dim in self.homology_dimensions_])
+        elif type(self.n_coefficients) in (list, tuple):
```

Comment:
this is not how we check the type.

it should be `elif isistance(self.n_coefficients, (list, tuple))`

You should use the modes Typing instead of the primitive types.